### PR TITLE
Add conditional checks on script parameters 

### LINKED ISSUE #203: Add conditional checks on script parameters
Hi team,

We saw that some playbooks or users pass wrong parameters to the scripts, we want to improve this by adding conditional checks when the scripts parse their parameters.
Examples:
https://github.com/wazuh/wazuh-packages/blob/master/rpms/generate_rpm_package.sh#L101
https://github.com/wazuh/wazuh-packages/blob/e31f2c76f7e659237bf2ed8a911d9d82bf4dd5db/rpms/generate_rpm_package.sh#L176

Thanks and regards.

### PR DESCRIPTION
**Pull request by DFolchA**
Hello team,
 This PR closes #203 By adding some checks to the generation scripts of:

- RPM
- DEB
- AIX
- SPLUNKAPP
- WAZUHAPP
- WPK

When parsing the arguments.

Regards,
Daniel Folch

### CODE REVIEW

File: aix/generate_wazuh_packages.sh [hunk 1]
```diff
@@ -223,9 +223,14 @@ main() {
         "-b"|"--branch")
           if [ -n "$2" ]
           then
-              wazuh_branch="$2"
-              build_rpm="yes"
-              shift 2
+              if [[ `curl https://api.github.com/repos/wazuh/wazuh-api/branches` =~ "$2" ]] || [[ `curl https://api.github.com/repos/wazuh/wazuh-api/tags` =~ "$2" ]]; then
```
Comment (BraulioV) [inline aix/generate_wazuh_packages.sh]: This check isn't necessary. `git` automatically checks this while cloning the branch.

File: debs/generate_debian_package.sh [hunk 2]
```diff
@@ -132,9 +132,14 @@ main() {
         case "$1" in
         "-b"|"--branch")
             if [ -n "$2" ]; then
-                BRANCH="$(echo $2 | cut -d'/' -f2)"
-                BUILD="yes"
-                shift 2
+                if [[ `curl https://api.github.com/repos/wazuh/wazuh-api/branches` =~ "$2" ]] || [[ `curl https://api.github.com/repos/wazuh/wazuh-api/tags` =~ "$2" ]]; then
```
Comment (BraulioV) [inline debs/generate_debian_package.sh]: This check isn't necessary. `git` automatically checks this while cloning the branch.

File: debs/generate_debian_package.sh [hunk 3]
```diff
@@ -144,16 +149,26 @@ main() {
...
+                fi
             else
                 help 1
             fi
             ;;
         "-a"|"--architecture")
             if [ -n "$2" ]; then
-                ARCHITECTURE="$2"
-                shift 2
+                if [[ "$2" == "x86_64" ]] || [[ "$2" == "amd64" ]]; then
```
Comment (BraulioV) [inline debs/generate_debian_package.sh]: This check should be done in the build function. In addition, this check is wrong. The script support amd64, i386 and ppc64le as architecture. Check it on the build function.

File: macos/generate_wazuh_packages.sh [hunk 4]
```diff
@@ -168,9 +168,14 @@ function main() {
         case "$1" in
         "-b"|"--branch")
             if [ -n "$2" ]; then
-                BRANCH_TAG="$2"
-                BUILD=yes
-                shift 2
+                if [[ `curl https://api.github.com/repos/wazuh/wazuh-api/branches` =~ "$2" ]] || [[ `curl https://api.github.com/repos/wazuh/wazuh-api/tags` =~ "$2" ]]; then
```
Comment (BraulioV) [inline macos/generate_wazuh_packages.sh]: This check isn't necessary. `git` automatically checks this while cloning the branch.

File: rpms/generate_rpm_package.sh [hunk 5]
```diff
@@ -153,9 +153,14 @@ main() {
         case "$1" in
         "-b"|"--branch")
             if [ -n "$2" ]; then
-                BRANCH="$2"
-                BUILD="yes"
-                shift 2
+                if [[ `curl https://api.github.com/repos/wazuh/wazuh-api/branches` =~ "$2" ]] || [[ `curl https://api.github.com/repos/wazuh/wazuh-api/tags` =~ "$2" ]]; then
```
Comment (BraulioV) [inline rpms/generate_rpm_package.sh]: This check isn't necessary. `git` automatically checks this while cloning the branch.

File: rpms/generate_rpm_package.sh [hunk 6]
```diff
@@ -165,16 +170,25 @@ main() {
...
+                fi
             else
                 help 1
             fi
             ;;
         "-a"|"--architecture")
             if [ -n "$2" ]; then
-                ARCHITECTURE="$2"
-                shift 2
+                if [[ "$2" == "x86_64" ]] || [[ "$2" == "amd64" ]]; then
```
Comment (BraulioV) [inline rpms/generate_rpm_package.sh]: This check should be done in the build function. In addition, this check is wrong. The script support x86_64 and i386. Check it on the build function.

File: splunkapp/generate_wazuh_splunk_app.sh [hunk 7]
```diff
@@ -88,15 +88,20 @@ main() {
...
             "-h"|"--help")
-                help 0 
+                help 0
             ;;
             "-b"|"--branch")
                 if [ -n "$2" ]; then
-                    BRANCH_TAG="$(echo $2 | cut -d'/' -f2)"
-                    HAVE_BRANCH=true
-                    shift 2
+                    if [[ `curl https://api.github.com/repos/wazuh/wazuh-api/branches` =~ "$2" ]] || [[ `curl https://api.github.com/repos/wazuh/wazuh-api/tags` =~ "$2" ]]; then
```
Comment (BraulioV) [inline splunkapp/generate_wazuh_splunk_app.sh]: This check isn't necessary. `git` automatically checks this while cloning the branch.

File: wazuhapp/generate_wazuh_app.sh [hunk 8]
```diff
@@ -71,9 +71,14 @@ main(){
         case "$1" in
         "-b"|"--branch")
             if [ -n "$2" ]; then
-                HAVE_BRANCH=true
-                BRANCH_TAG="$(echo "$2" | cut -d "/" -f2)"
-                shift 2
+                if [[ `curl https://api.github.com/repos/wazuh/wazuh-api/branches` =~ "$2" ]] || [[ `curl https://api.github.com/repos/wazuh/wazuh-api/tags` =~ "$2" ]]; then
```
Comment (BraulioV) [inline wazuhapp/generate_wazuh_app.sh]: This check isn't necessary. `git` automatically checks this while cloning the branch.

File: wpk/generate_wpk_package.sh [hunk 9]
```diff
@@ -101,20 +101,30 @@ function main() {
   do
       case "$1" in
       "-t"|"--target-system")
-          if [[ -n "$2" ]] && [[ "$2" == "linux" || "$2" == "windows" ]]; then
-            local TARGET="$2"
-            local HAVE_TARGET=true
-            shift 2
+          if [[ -n "$2" ]]; then
```
Comment (BraulioV) [inline wpk/generate_wpk_package.sh]: This change isn't necessary.

File: wpk/generate_wpk_package.sh [hunk 10]
```diff
@@ -101,20 +101,30 @@ function main() {
...
             echo "ERROR: Missing target system."
             help 1
           fi
           ;;
       "-b"|"--branch")
           if [[ -n "$2" ]]; then
-              local BRANCH="$(echo $2 | cut -d'/' -f2)"
-              local HAVE_BRANCH=true
-              shift 2
+            if [[ `curl https://api.github.com/repos/wazuh/wazuh-api/branches` =~ "$2" ]] || [[ `curl https://api.github.com/repos/wazuh/wazuh-api/tags` =~ "$2" ]]; then
```
Comment (BraulioV) [inline wpk/generate_wpk_package.sh]: This check isn't necessary. `git` automatically checks this while cloning the branch.

Comment (DFolchA) [reply wpk/generate_wpk_package.sh]: Done in https://github.com/wazuh/wazuh-packages/pull/255/commits/ac22541a388ac6ed3a45b864fe89d04baccd4d20

File: wpk/generate_wpk_package.sh (hunk 9 above)
Comment (DFolchA) [reply wpk/generate_wpk_package.sh]: Done in https://github.com/wazuh/wazuh-packages/pull/255/commits/ac22541a388ac6ed3a45b864fe89d04baccd4d20

File: wazuhapp/generate_wazuh_app.sh (hunk 8 above)
Comment (DFolchA) [reply wazuhapp/generate_wazuh_app.sh]: Done in https://github.com/wazuh/wazuh-packages/pull/255/commits/ac22541a388ac6ed3a45b864fe89d04baccd4d20

File: splunkapp/generate_wazuh_splunk_app.sh (hunk 7 above)
Comment (DFolchA) [reply splunkapp/generate_wazuh_splunk_app.sh]: Done in https://github.com/wazuh/wazuh-packages/pull/255/commits/ac22541a388ac6ed3a45b864fe89d04baccd4d20

File: rpms/generate_rpm_package.sh (hunk 6 above)
Comment (DFolchA) [reply rpms/generate_rpm_package.sh]: Done in https://github.com/wazuh/wazuh-packages/pull/255/commits/ac22541a388ac6ed3a45b864fe89d04baccd4d20

File: rpms/generate_rpm_package.sh (hunk 5 above)
Comment (DFolchA) [reply rpms/generate_rpm_package.sh]: Done in https://github.com/wazuh/wazuh-packages/pull/255/commits/ac22541a388ac6ed3a45b864fe89d04baccd4d20

File: macos/generate_wazuh_packages.sh (hunk 4 above)
Comment (DFolchA) [reply macos/generate_wazuh_packages.sh]: Done in https://github.com/wazuh/wazuh-packages/pull/255/commits/ac22541a388ac6ed3a45b864fe89d04baccd4d20

File: debs/generate_debian_package.sh (hunk 3 above)
Comment (DFolchA) [reply debs/generate_debian_package.sh]: Done in https://github.com/wazuh/wazuh-packages/pull/255/commits/ac22541a388ac6ed3a45b864fe89d04baccd4d20

File: debs/generate_debian_package.sh (hunk 2 above)
Comment (DFolchA) [reply debs/generate_debian_package.sh]: Done in https://github.com/wazuh/wazuh-packages/pull/255/commits/ac22541a388ac6ed3a45b864fe89d04baccd4d20

File: aix/generate_wazuh_packages.sh (hunk 1 above)
Comment (DFolchA) [reply aix/generate_wazuh_packages.sh]: Done in https://github.com/wazuh/wazuh-packages/pull/255/commits/ac22541a388ac6ed3a45b864fe89d04baccd4d20

File: debs/generate_debian_package.sh [hunk 11]
```diff
@@ -152,8 +152,13 @@ main() {
             ;;
         "-a"|"--architecture")
             if [ -n "$2" ]; then
-                ARCHITECTURE="$2"
-                shift 2
+                if [[ "$2" == "i386" ]] || [[ "$2" == "amd64" ]|| [[ "$2" == "amd64" ]]; then
```
Comment (BraulioV) [inline debs/generate_debian_package.sh]: The check amd64 is duplicated. It should consider x86_64 and ppc64le.

File: debs/generate_debian_package.sh [hunk 12]
```diff
@@ -152,8 +152,13 @@ main() {
...
+                    ARCHITECTURE="$2"
+                    shift 2
+                else
+                    echo "Architecture must be x86_64 or amd64"
```
Comment (BraulioV) [inline debs/generate_debian_package.sh]: This message must be changed.

File: rpms/generate_rpm_package.sh [hunk 13]
```diff
@@ -173,8 +173,12 @@ main() {
             ;;
         "-a"|"--architecture")
             if [ -n "$2" ]; then
-                ARCHITECTURE="$2"
-                shift 2
+                if [[ "$2" == "x86_64" ]] || [[ "$2" == "i386" ]]; then
```
Comment (BraulioV) [inline rpms/generate_rpm_package.sh]: This check must consider amd64 also

File: rpms/generate_rpm_package.sh (hunk 6 above)
Comment (BraulioV) [reply rpms/generate_rpm_package.sh]: This isn't resolved yet.

File: debs/generate_debian_package.sh (hunk 3 above)
Comment (BraulioV) [reply debs/generate_debian_package.sh]: This isn't resolved yet.

File: rpms/generate_rpm_package.sh (hunk 13 above)
Comment (DFolchA) [reply rpms/generate_rpm_package.sh]: Done in https://github.com/wazuh/wazuh-packages/pull/255/commits/7227907bae4d24ba9920560ff50f6c40e0a7dc8f

**Pull request merged**